app/celery/letters_pdf_tasks.py

```diff
@@ -1,424 +0,0 @@
-    )
-    resp.raise_for_status()
-
-    pages_per_sheet = 2
-    billable_units = math.ceil(int(resp.headers.get('X-pdf-page-count', 0)) / pages_per_sheet)
-
-    return resp.content, billable_units
-
-
-@notify_celery.task(name='collate-letter-pdfs-for-day')
-@cronitor('collate-letter-pdfs-for-day')
-def collate_letter_pdfs_for_day(date=None):
-    if not date:
-        # Using the truncated date is ok because UTC to BST does not make a difference to the date,
-        # since it is triggered mid afternoon.
-        date = datetime.utcnow().strftime('%Y-%m-%d')
-
-    letter_pdfs = sorted(
-        s3.get_s3_bucket_objects(current_app.config['LETTERS_PDF_BUCKET_NAME'], subfolder=date),
-        key=lambda letter: letter['Key'],
-    )
-    for i, letters in enumerate(group_letters(letter_pdfs)):
-        filenames = [letter['Key'] for letter in letters]
-
-        hash = urlsafe_b64encode(sha512(''.join(filenames).encode()).digest())[:20].decode()
-        # eg NOTIFY.2018-12-31.001.Wjrui5nAvObjPd-3GEL-.ZIP
-        dvla_filename = 'NOTIFY.{date}.{num:03}.{hash}.ZIP'.format(date=date, num=i + 1, hash=hash)
-
-        current_app.logger.info(
-            'Calling task zip-and-send-letter-pdfs for %d pdfs to upload %s with total size %s bytes',
-            len(filenames),
-            dvla_filename,
-            '{:,}'.format(sum(letter['Size'] for letter in letters)),
-        )
-        notify_celery.send_task(
-            name=TaskNames.ZIP_AND_SEND_LETTER_PDFS,
-            kwargs={'filenames_to_zip': filenames, 'upload_filename': dvla_filename},
-            queue=QueueNames.PROCESS_FTP,
-            compression='zlib',
-        )
-
-
-def group_letters(letter_pdfs):
-    """
-    Group letters in chunks of MAX_LETTER_PDF_ZIP_FILESIZE. Will add files to lists, never going over that size.
-    If a single file is (somehow) larger than MAX_LETTER_PDF_ZIP_FILESIZE that'll be in a list on it's own.
-    If there are no files, will just exit (rather than yielding an empty list).
-    """
-    running_filesize = 0
-    list_of_files = []
-    for letter in letter_pdfs:
-        if letter['Key'].lower().endswith('.pdf') and letter_in_created_state(letter['Key']):
-            if (
-                running_filesize + letter['Size'] > current_app.config['MAX_LETTER_PDF_ZIP_FILESIZE']
-                or len(list_of_files) >= current_app.config['MAX_LETTER_PDF_COUNT_PER_ZIP']
-            ):
-                yield list_of_files
-                running_filesize = 0
-                list_of_files = []
-
-            running_filesize += letter['Size']
-            list_of_files.append(letter)
-
-    if list_of_files:
-        yield list_of_files
-
-
-def letter_in_created_state(filename):
-    # filename looks like '2018-01-13/NOTIFY.ABCDEF1234567890.D.2.C.C.20180113120000.PDF'
-    subfolder = filename.split('/')[0]
-    ref = get_reference_from_filename(filename)
-    notifications = dao_get_notifications_by_references([ref])
```

Comment:
I believe `dao_get_notifications_by_references` can be removed in `app.dao.notifications_dao.dao_get_notifications_by_references.py`. It is only used here and in testing